### PR TITLE
Handle http/2 request body without content-length

### PR DESCRIPTION
**Pull request by hamnis**
Http/2 is always streaming, so setting content-length is optional.

### CODE REVIEW

File: core/src/main/scala/org/http4s/jdkhttpclient/JdkHttpClient.scala [hunk 1]
```diff
@@ -63,10 +63,15 @@ object JdkHttpClient {
     def convertRequest(req: Request[F]): Resource[F, HttpRequest] =
       flow.toPublisher(req.body.chunks.map(_.toByteBuffer)).evalMap { publisher =>
         convertHttpVersionFromHttp4s[F](req.httpVersion).map { version =>
+          def isStreaming = version match {
+            case HttpClient.Version.HTTP_1_1 => req.isChunked
+            case HttpClient.Version.HTTP_2 => req.contentLength.isEmpty
```
Comment (danicheg) [inline core/src/main/scala/org/http4s/jdkhttpclient/JdkHttpClient.scala:68]: At first glance, this seems a little tricky. But given the streaming nature, I think relying on `req.body eq Stream.empty[Nothing]` is hardly a better option.

File: core/src/main/scala/org/http4s/jdkhttpclient/JdkHttpClient.scala [hunk 2]
```diff
@@ -63,10 +63,15 @@ object JdkHttpClient {
     def convertRequest(req: Request[F]): Resource[F, HttpRequest] =
       flow.toPublisher(req.body.chunks.map(_.toByteBuffer)).evalMap { publisher =>
         convertHttpVersionFromHttp4s[F](req.httpVersion).map { version =>
+          def isStreaming = version match {
```
Comment (danicheg) [inline core/src/main/scala/org/http4s/jdkhttpclient/JdkHttpClient.scala:66]: To reduce confusion, I'd suggest renaming it. One might assume that `HTTP/2` mandates the streamed body or the presence of `Content-Length` to distinguish that the body is not streamed (while the streamed body alongside `Content-Length` is ok in HTTP/2). IMHO, something implying `consumeFully` might be fine.
```suggestion
          def consumeFully = version match {
```

**Review by danicheg — COMMENTED** [review]
Not blocking progress, just sharing some food for thought.

File: core/src/main/scala/org/http4s/jdkhttpclient/JdkHttpClient.scala [hunk 3]
```diff
@@ -63,16 +64,24 @@ object JdkHttpClient {
...
                 BodyPublishers.fromPublisher(publisher)
               else
                 req.contentLength match {
                   case Some(length) if length > 0L =>
                     BodyPublishers.fromPublisher(publisher, length)
-                  case _ => BodyPublishers.noBody
+                  case _ =>
+                    // If we dont do this, we might block finalization
+                    publisher.subscribe(DrainingSubscriber)
+                    BodyPublishers.noBody
```
Comment (danicheg) [inline core/src/main/scala/org/http4s/jdkhttpclient/JdkHttpClient.scala:84]: 👍🏻

**Review by samspills — APPROVED** [review]

**Pull request merged**